src/runtime-pmi.c

```diff
@@ -78,11 +81,13 @@ decode(const char *inval, void *outval, size_t outvallen)
     size_t i;
     char *ret = (char*) outval;
 
-    if (outvallen != strlen(inval) / 2) {
+    size_t outlen = strlen(inval) / 2;
+
+    if (outvallen < outlen) {
```

Comment:
(A long long time ago) we discovered a use-case where `shmem_runtime_get` does not know the length/size of the incoming value (i.e., the nodename/hostname string).  I believe this change to the `outvallen` check was necessary with the original approach (in #653), where `shmem_runtime_put` was called just once.  But now I put the hostname string-length into the KVS 1<sup>st</sup>, then the string itself 2<sup>nd</sup>, which actually avoids that problem.

Anyway, I'm fine removing this change if you want because it's not needed anymore.  But I was also thinking it's a relatively benign change (as far as I can tell), and it adds this new capability to the KVS.

Reply:
I ended up reverting this change.